source/declarations.tex

```diff
@@ -1896,29 +1896,35 @@
 
 \pnum
 A type \tcode{T} containing a placeholder type,
-and a corresponding initializer $E$,
+and a corresponding \grammarterm{initializer-clause} $E$,
 are determined as follows:
 \begin{itemize}
 \item
-for a non-discarded \tcode{return} statement that occurs
+For a non-discarded \tcode{return} statement that occurs
 in a function declared with a return type
 that contains a placeholder type,
 \tcode{T} is the declared return type
-and $E$ is the operand of the \tcode{return} statement.
+and $E$ is the operand of the \tcode{return} statement;
+if the operand is an \grammarterm{expression} $X$
+that is not an \grammarterm{assignment-expression}\iref{expr.comma},
+$E$ is \tcode{($X$)}.
 If the \tcode{return} statement
 has no operand,
-then $E$ is \tcode{void()};
+then $E$ is \tcode{void()}.
 \item
-for a variable declared with a type
+For a variable declared with a type
 that contains a placeholder type,
-\tcode{T} is the declared type of the variable
-and $E$ is the initializer.
+\tcode{T} is the declared type of the variable.
+If the initializer of the variable is a \grammarterm{brace-or-equal-initializer}
+of the form \tcode{= \grammarterm{initializer-clause}},
+$E$ is the \grammarterm{initializer-clause}.
 If the initialization is direct-list-initialization,
 the initializer shall be a \grammarterm{braced-init-list}
 containing only a single \grammarterm{assignment-expression}
-and $E$ is the \grammarterm{assignment-expression};
+and $E$ is the \grammarterm{assignment-expression}.
+Otherwise, $E$ is the initializer.
```

Comment:
There's more `decltype(auto)` awkwardness here; I think this does the wrong thing for:
```
int a;
decltype(auto) b(a);
```
... which I think should deduce the type of `b` as `int`, but if we take `E` as being `(a)` then we presumably reinterpret it as a parenthesized expression, resulting in the type of `b` being `int&`.

Reply:
Would it make sense to move [dcl.spec.auto.general]/4's "If the initializer is a parenthesized expression-list, the expression-list shall be a single assignment-expression." here, and then modify it to match the treatment of the braced-init-list case?

Reply:
Yes, that makes a lot of sense.

I've also integrated the two sentences about constraints on return type deduction that followed the bulleted list. It all looks clean and sane now, except for the wrinkle about forming a parenthesized expression when we start out with a comma expression. This has now been highlighted in a note.

Reply:
@zygoloid: PTAL